internal/mode/static/telemetry/job.go

```diff
@@ -0,0 +1,56 @@
+package telemetry
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+)
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . DataCollector
+
+// DataCollector collects telemetry data.
+type DataCollector interface {
+	// Collect collects and returns telemetry Data.
+	Collect(ctx context.Context) (Data, error)
+}
+
+//go:generate go run github.com/maxbrunsfeld/counterfeiter/v6 . HealthChecker
+
+// HealthChecker checks if the NGF Pod is ready.
+type HealthChecker interface {
+	// GetReadyIfClosedChannel returns a channel which determines if the NGF Pod is ready.
+	GetReadyIfClosedChannel() <-chan struct{}
+}
+
+func CreateTelemetryJobWorker(
```

Comment:
@sjberman In order to test the worker function, I decided to refactor this function to purely create the worker, thus I could plug it into the CronJob and test it exactly how Job was tested in `job_test`. Thoughts?

Reply:
I think it's fine to have the worker be defined here so you can inject it into the job elsewhere. However, I don't think you need to actually test the job logic, because that's what `job_test` does. Just test that the worker function itself does what's expected.

Also, I don't think this file needs to be named `job.go` anymore, since it no longer contains that logic.

Reply:
In other words, we don't need to start the job and check for X number of calls to the worker, because we already test that for the generic case. The worker-specific test should just test the worker logic.